add-LTI-dev/add-LTI-dev.sh

```diff
@@ -28,15 +30,52 @@ read -r -a client_ids_array <<< "${client_ids}"
 # Check that the host is up
 ../check-up/check-up "https://${host}/help_links" || (echo "Not running, host isn't up"; exit 1)
 
-# Add each tool where we have a client ID
+
+# -----------------------------------------------------------------
+
+# Get a list of all external tools & extract the line showing the developer_key_ids, save in a tmp file
+
+curl -s -X GET https://${host}/api/v1/accounts/1/external_tools?per_page=1000  \
+    -H "Authorization: Bearer ${token}"  | jq | grep "developer_key_id" > /tmp/dev_key_ids.txt
+
+# Add each tool where we have a client ID but only if it isnt already installed
 for client_id in "${client_ids_array[@]}"
 do
-  printf "\nSetting up tool with client ID of ${client_id}: "
-  curl -X POST https://${host}/api/v1/accounts/1/external_tools  \
-    -H "Authorization: Bearer ${token}" \
-    -F "client_id=${client_id}" | jq -r '.name' | tr '\n' ' '; \
-    printf "successfully set up\n\n" \
-    || printf "Unable to set tool with client ID ${client_id}. Please do this manually.\n\n"
-    printf "\n\n"
+
+  # generate the developer_key_id, remove the 122010000... bit
+  tmp="${client_id:5}"
+  developer_key_id="$((10#$tmp))"
```

Comment:
What is this doing?

Reply:
cunningly extracting the tool ID from the 12201000...... identifier. It lops off the first 5 characters then dumps all the leading zeroes.